### PR TITLE
core-to-plc: support mutually recursive (monomorphic) functions

### PR DESCRIPTION
**Pull request by michaelpj**
Pleasingly straightforward now we have `fixN`.

We can't do polymorphic functions because our fixpoint operators don't currently handle polymorphic functions.

### CODE REVIEW

File: core-to-plc/src/Language/Plutus/CoreToPLC.hs [hunk 1]
```diff
@@ -760,30 +735,16 @@ The answer is simple - we pass them through as a tuple.
...
-    tuple
-    (\a_1 ->
-      ..
-      tuple_i b_1 .. b_i
-    )
-  )
-)
+into this:
+
+($fixN n$ (\choose f1 ... fn . choose b_1 ... b_i))
```
Comment (j-mueller) [inline core-to-plc/src/Language/Plutus/CoreToPLC.hs]: *n = i*, correct? why the two indices?

Comment (michaelpj) [reply core-to-plc/src/Language/Plutus/CoreToPLC.hs]: typo

File: core-to-plc/test/IllTyped.hs [hunk 2]
```diff
@@ -39,3 +47,28 @@ ptreeConstruct = plc (Two (Two (One ((1,2),(3,4)))) :: B Int)
...
+-- not using case to avoid literal cases
+fib = plc (let fib :: Int -> Int
+               fib n = if n == 0 then 0 else if n == 1 then 1 else fib(n-1) + fib(n-2)
+           in fib)
+
+sumDirect :: PlcCode
+sumDirect = plc (let sum :: [Int] -> Int
+                     sum []     = 0
+                     sum (x:xs) = x + sum xs
+                 in sum)
```
Comment (effectfully) [inline core-to-plc/test/IllTyped.hs]: Why does this required mutual recursion?

Comment (michaelpj) [reply core-to-plc/test/IllTyped.hs]: It doesn't! But it uses the same machinery. I meant to add an even/odd example, I'll do that.

Comment (michaelpj) [reply core-to-plc/test/IllTyped.hs]: Added.

**Pull request merged**